litgpt/chat/base.py

```diff
@@ -120,11 +120,18 @@ def decode(fabric: L.Fabric, tokenizer: Tokenizer, token_stream: Iterator[torch.
     return tokens_generated
 
 
-def process_prompt(prompt, model, tokenizer, prompt_style, fabric, temperature, top_k, top_p, stop_tokens):
+def process_prompt(prompt, model, tokenizer, prompt_style, fabric, temperature, max_new_tokens, top_k, top_p, stop_tokens):
     prompt = prompt_style.apply(prompt=prompt)
     encoded_prompt = tokenizer.encode(prompt, device=fabric.device)
+
+    first_turn = model.mask_cache is None
+    max_returned_tokens = encoded_prompt.size(0) + max_new_tokens
+    if first_turn or max_returned_tokens > model.max_seq_length:
```

Comment:
```suggestion
    if first_turn or max_returned_tokens > model.max_seq_length:
```

Maybe it's just an early Monday morning for me ... but shouldn't it be flipped because otherwise you might end up exceeding the model's supported context length.